### ISSUE TITLE
Erasing entire content of `TextBox` doesn't call `set`-accessor of bound `Text` property if property always returns `string.Empty`

### ISSUE THREAD

**Issue by duck-dev** [AvaloniaUI/Avalonia#6879]
**Describe the bug**
Erasing the last remaining character in a `TextBox` doesn't call the `set`-accessor of the property that's bound to the `Text` property of the `TextBox` element if the property always returns `string.Empty`, because I want the content of the `TextBox` to be an empty string at the beginning. While adding a character or erasing any other letter that's not the last remaining one works as expected (the `set`-accessor does get called), erasing the last remaining character and therefore emptying the `TextBox` content doesn't call the `set`-accessor of the bound property in case the property always returns `string.Empty`.

**To Reproduce**
XAML:
```xaml
<Design.DataContext>
    <addPages:SomeViewModel/>
</Design.DataContext>

<TextBox Text="{Binding SomeProperty}" Width="100" Height="60" Watermark="Something" MaxLength="5"
                TextAlignment="Center" HorizontalContentAlignment="Center" HorizontalAlignment="Center"
                VerticalAlignment="Center" Foreground="Black" FontWeight="Bold" FontSize="30"/>
```

C# of the ViewModel, respectively the class the ViewModel is directly inheriting in my specific case:
```cs
protected string? SomeProperty
{
    get => string.Empty;
    set => System.Diagnostics.Trace.WriteLine("Set-accessor is getting called!!!!!");
}
```

1. Now open the Debug Output and watch the logs as you change the content of the `TextBox`
2. Enter a random character inside the `TextBox`
3. Erase it again. You should see only one log. My expected behaviour would be two logs, but that doesn't happen in any case.

**Expected behavior**
The `set`-accessor of the bound property should be called when the last remaining character is getting erased as well, as it does when you add a new character or erase a character that's not the last remaining one, even if the property returns `string.Empty`.

**Desktop (please complete the following information):**
- OS: Linux (Pop! OS 21.04)]
- Avalonia Version: 0.10.3

**Other remarks:**
Please let me know if there is an easier way to let the default text (at the beginning, before it is edited) be an empty string, so the watermark can be seen. I'm sorry if I'm overcomplicating things, but I'm quite tired and can't think of a better solution to solve this little issue.

**Comment by ahopper** [comment]
I would think this is expected behaviour as the framework won't consider it a change as the value is the same as that returned by get.
Simply defaulting the value of the property to empty should do what you want.

**Comment by duck-dev** [comment]
@ahopper I suspect that as well, but it is not possible to assign a value to a non-auto-property. The C# side however isn't what's bothering me. The main reason why I asked is to understand the behaviour of Avalonia in the background. Although I can't really come up with an elegant solution that's working as expected at the same time.

**Comment by danwalmsley** [comment]
Bindings need to be able to read the actual value back in order to work correctly. What is the use case scenario for not allowing the getter to return the true value? Most likely what your trying to do with that there are different patterns to acheive it.

BTW you may want to do: {Binding Mode=OneWayToSource}, and that might give you the desired behavior in that case.

**Comment by duck-dev** [comment]
I'm just experimenting around with this "issue" after I figured out that returning `string.Empty` doesn't match well with such a binding. It was more of a why, than a how to fix it issue. And initially I thought it might be an issue in the framework, since I didn't notice I always returned `string.Empty`, which turned out as an accident after I noticed it.

Thank you very much for the guidance though. I seem to understand better how Bindings blend with the properties in the background, which was the intention after I figured out why this was happening.
